kernel/kernel-api/src/test/scala/io/delta/kernel/internal/util/SchemaUtilsSuite.scala

```diff
@@ -841,58 +702,275 @@ class SchemaUtilsSuite extends AnyFunSuite {
           fieldMetadata(id = 1, physicalName = "id"))),
     // Map with struct key renamed
     (
-      new StructType()
-        .add(
-          "id_map",
-          new MapType(
-            new StructType()
-              .add("id", IntegerType.INTEGER, false, fieldMetadata(id = 2, physicalName = "id")),
-            IntegerType.INTEGER,
-            false),
-          true,
-          fieldMetadata(id = 1, physicalName = "id_map")),
-      new StructType()
-        .add(
-          "id_map",
-          new MapType(
-            new StructType()
-              .add(
-                "renamed_id",
-                IntegerType.INTEGER,
-                false,
-                fieldMetadata(id = 2, physicalName = "id")),
-            IntegerType.INTEGER,
-            false),
-          true,
-          fieldMetadata(id = 1, physicalName = "id_map"))),
-    // Struct with array of struct field where inner struct is missing physical name
+      mapWithStructKey,
+      mapWithStructKey(
+        new MapType(
+          new StructType()
+            .add(
+              "renamed_id",
+              IntegerType.INTEGER,
+              true,
+              fieldMetadata(id = 2, physicalName = "id")),
+          IntegerType.INTEGER,
+          false),
+        fieldMetadata(id = 1, physicalName = "map"))),
+    // Struct with array of struct field where inner struct field is renamed
     (
+      structWithArrayOfStructs,
       new StructType()
         .add(
           "top_level_struct",
           new StructType().add(
             "array",
             new ArrayType(
-              new StructType().add("id", IntegerType.INTEGER, fieldMetadata(4L, "id")),
+              new StructType().add("renamed_id", IntegerType.INTEGER, fieldMetadata(4, "id")),
               false),
             false,
-            fieldMetadata(2L, "array_field")),
-          fieldMetadata(1L, "top_level_struct")),
+            fieldMetadata(2, "array_field")),
+          fieldMetadata(1, "top_level_struct"))))
+
+  test("validateUpdatedSchema succeeds with valid ID and physical name") {
+    forAll(validUpdatedSchemas) { (schemaBefore, schemaAfter) =>
```

Comment:
Can you add a case where we just change something in the metadata? Like a column comment? (maybe nested & unnested)

Reply:
Sure, I can do that!